src/topojsonm/topojsonm.ml

```diff
@@ -1 +1,227 @@
+    | `Lexeme l -> l
+    | `Error e -> raise (Abort (`Error (loc (), e)))
+    | `End -> raise (Abort `EOI)
+    | `Await -> assert false
+  in
+  let rec enter l stack =
+    match l with
+    | `Os -> obj [] stack
+    | _ -> raise (Abort (`Unexpected "decoding single object failed"))
+  and value l stack =
+    match l with
+    | `Os -> obj [] stack
+    | `As -> arr [] stack
+    | (`Null | `Bool _ | `String _ | `Float _) as l -> continue l stack
+    | _ -> raise (Abort (`Unexpected "value"))
+  and arr so_far stack =
+    match dec () with
+    | `Ae -> continue (`A (List.rev so_far)) stack
+    | l ->
+        let stack = Stack.In_array (so_far, stack) in
+        value l stack
+  and obj so_far stack =
+    match dec () with
+    | `Oe -> continue (`O (List.rev so_far)) stack
+    | `Name n ->
+        let stack = Stack.In_object (n, so_far, stack) in
+        value (dec ()) stack
+    | _ -> raise (Abort (`Unexpected "object fields"))
+  and continue v stack =
+    match stack with
+    | Stack.In_array (vs, stack) ->
+        let so_far = v :: vs in
+        arr so_far stack
+    | Stack.In_object (n, ms, stack) ->
+        let so_far = (n, v) :: ms in
+        obj so_far stack
+    | Stack.Empty -> v
+  in
+  enter (dec ()) Empty
+
+let encode_value e json =
+  let module Stack = struct
+    type t =
+      | In_array of Ezjsonm.value list * t
+      | In_object of (string * Ezjsonm.value) list * t
+      | Empty
+  end in
+  let enc e l = ignore (Jsonm.encode e (`Lexeme l)) in
+  let rec t v e stack =
+    match v with
+    | `A vs ->
+        enc e `As;
+        arr vs e stack
+    | `O ms ->
+        enc e `Os;
+        obj ms e stack
+  and value v e stack =
+    match v with
+    | (`Null | `Bool _ | `Float _ | `String _) as v ->
+        enc e v;
+        continue e stack
+    | #Ezjsonm.t as x -> t (x :> Ezjsonm.t) e stack
+  and arr vs e stack =
+    match vs with
+    | v :: vs' ->
+        let stack = Stack.In_array (vs', stack) in
+        value v e stack
+    | [] ->
+        enc e `Ae;
+        continue e stack
+  and obj ms e stack =
+    match ms with
+    | (n, v) :: ms ->
+        enc e (`Name n);
+        let stack = Stack.In_object (ms, stack) in
+        value v e stack
+    | [] ->
+        enc e `Oe;
+        continue e stack
+  and continue e stack =
+    match stack with
+    | Stack.In_array (vs, stack) -> arr vs e stack
+    | Stack.In_object (ms, stack) -> obj ms e stack
+    | Stack.Empty -> ()
+  in
+  value json e Stack.Empty
+
+let map_object f src dst =
+  let decoder = Jsonm.decoder src in
+  let encoder = Jsonm.encoder dst in
+  let loc () = Jsonm.decoded_range decoder in
+  let enc v =
+    match Jsonm.encode encoder v with
+    | `Ok -> ()
+    | `Partial -> raise (Abort (`Unexpected "partial encoding"))
+  in
+  let rec go () =
+    match Jsonm.decode decoder with
+    | `Lexeme (`Name "object" as t) ->
+        (match Jsonm.decode decoder with
+        | `Lexeme (`Name "example" as ex) -> (
```

Comment:
We don't want to _only_ pull out one particular case in the `objects` field, we want to iterate over all of them providing the function `f` with both the name associated with the object _and_ the decoded object converted to the Topojson value.